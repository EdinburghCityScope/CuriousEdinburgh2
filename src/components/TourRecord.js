import React, { Component } from 'react';
import { Modal,
         Text,
         ScrollView,
         StyleSheet,
         TouchableHighlight,
         View } from 'react-native';

const styles = StyleSheet.create({
    page: {
        flex: 1, justifyContent: 'space-between',
    },
    header: {
        flexDirection: 'row',
    },
    body: {

    },
    image: {
        width: 100,
        height: 100,
        borderRadius: 10,
    },
});

export default class TourRecord extends Component {
    constructor(props) {
        super(props);

        this.state = {
            visible: false,
            record: {
                images: [],
            },
        };
    }

    show(rec) {
        // console.log(record);
        this.setState({
            visible: true,
            record: rec,
        });
    }

    close() {
        this.setState({ visible: false });
    }

    render() {
        return (
          <Modal
            style={styles.page}
            transparent={false}
            visible={this.state.visible}
            onRequestClose={() => { }}
          >
            <View>
              <View
                style={
                    styles.header
                }
              >
                <TouchableHighlight
                  onPress={() => { this.close(); }}
                >
                  <Text>Close</Text>
                </TouchableHighlight>
                <Text>{this.state.record.title}</Text>
              </View>
            </View>
            <View>
              <Text>Hello</Text>
              <ScrollView />
              <Text>{this.state.record.description}</Text>
            </View>
          </Modal>
        );
    }
}
