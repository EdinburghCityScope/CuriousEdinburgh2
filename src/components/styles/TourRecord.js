import { StyleSheet } from 'react-native';

module.exports = StyleSheet.create({
    page: {

    },
    header: {
        flexDirection: 'row',
        height: 60,
        backgroundColor: '#268da8',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    left: {
        alignSelf: 'center',
        top: 2,
        padding: 4,
    },
    close: {
        backgroundColor: '#268da8',
        color: 'white',
    },
    title: {
        fontSize: 20,
        color: 'white',
        fontWeight: 'bold',
    },
    right: {
        //flex: 1,
    },
    images: {
        flexDirection: 'row',
        padding: 20,
    },
    imageContainer: {
        paddingRight: 10,
    },
    image: {
        width: 200,
        height: 200,
        borderRadius: 10,
        paddingRight: 10,
    },
    details: {
        borderWidth: 10,
        borderColor: '#efeff4',
    },
    address: {
        color: '#679fad',
        fontSize: 20,
        fontWeight: 'bold',
        padding: 10,
    },
    description: {
        paddingTop: 10,
        fontSize: 18,
        padding: 10,
    },
    linksTitle: {
        fontSize: 30,
        fontWeight: 'bold',
        padding: 10,
    },
    link: {
        color: '#679fad',
        fontSize: 20,
        fontWeight: 'bold',
        padding: 10,
    },
});
