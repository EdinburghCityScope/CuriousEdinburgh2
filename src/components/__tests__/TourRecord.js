import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import TourRecord from '../TourRecord';

test('renders correctly', () => {
    const TITLE = 'test record';
    const VIDEO = 'https://www.youtube.com/watch?v=i64wvdnHSIw';
    const IMG1 = 'https://image1.png';
    const IMG2 = 'https://image2.png';
    const L1 = 'https://link1.html';
    const L2 = 'https://link2.html';
    const L3 = 'https://link3.html';

    const record = {
        title: TITLE,
        images: [IMG1, IMG2],
        additionalLinks: [L1, L2, L3],
        video: VIDEO,
    };

    jest.mock('WebView', () => {
        const RealComponent = require.requireActual('WebView');
        // eslint-disable-next-line
        const MockReact = require('React');
        class WebView extends MockReact.Component {
            render() {
                if (!this.props) {
                    return null;
                }
                return MockReact.createElement('WebView', this.props, this.props.children);
            }
        }
        WebView.propTypes = RealComponent.propTypes;
        return WebView;
    });

    const tr = <TourRecord ref={(c) => { this.modal = c; }} />;

    const component = renderer.create(tr);
    this.modal.show(record);
    expect(component).toMatchSnapshot();

    const compJSON = component.toJSON();

    // header
    const header = compJSON.children[1];
    expect(header.children[0].children[1].children[0]).toEqual(TITLE);

    const body = compJSON.children[2];

    // there are 3 pieces of media
    const media = body.children[0];
    expect(media.children.length).toEqual(3);

    const video = media.children[0];
    expect(video.children[0].props.source.uri).toEqual(VIDEO);

    const firstImage = media.children[1];
    expect(firstImage.children[0].props.source.uri).toEqual(IMG1);

    const secondImage = media.children[2];
    expect(secondImage.children[0].props.source.uri).toEqual(IMG2);

    // associated links
    const links = body.children[1].children[3];
    expect(links.children[0].children[0].children[0]).toEqual(L1);
    expect(links.children[1].children[0].children[0]).toEqual(L2);
    expect(links.children[2].children[0].children[0]).toEqual(L3);
});
