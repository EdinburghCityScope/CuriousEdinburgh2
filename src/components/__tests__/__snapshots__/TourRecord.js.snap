exports[`test renders correctly 1`] = `
<Modal
  onRequestClose={[Function]}
  style={undefined}
  transparent={false}
  visible={true}>
  <Modal
    onRequestClose={[Function]}
    visible={false}>
    <ScrollView
      centerContent={true}
      contentContainerStyle={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
      maximumZoomScale={3}
      minimumZoomScale={0.5}>
      <Image
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        maximumZoomScale={3}
        minimumZoomScale={0.5}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        onTap={[Function]}
        resizeMode="contain"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": 1334,
            "width": 750,
          }
        }
        testID={undefined} />
    </ScrollView>
  </Modal>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#268da8",
          "flexDirection": "row",
          "height": 60,
          "justifyContent": "space-between",
        }
      }>
      <View
        style={
          Object {
            "alignSelf": "center",
            "padding": 5,
            "top": 4,
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 40,
                },
                Object {
                  "backgroundColor": "#268da8",
                  "color": "white",
                },
                Object {
                  "fontFamily": "FontAwesome",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }>
            ï‹“
          </Text>
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 20,
            "fontWeight": "bold",
          }
        }>
        test record
      </Text>
      <View
        style={undefined} />
    </View>
  </View>
  <ScrollView
    style={undefined}>
    <ScrollView
      horizontal={true}
      style={
        Object {
          "flexDirection": "row",
          "padding": 20,
        }
      }>
      <View
        style={
          Object {
            "paddingRight": 10,
          }
        }>
        <WebView
          source={
            Object {
              "uri": "https://www.youtube.com/watch?v=i64wvdnHSIw",
            }
          }
          style={
            Object {
              "borderRadius": 10,
              "height": 200,
              "width": 200,
            }
          } />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "paddingRight": 10,
            },
          ]
        }
        testID={undefined}>
        <Image
          source={
            Object {
              "uri": "https://image1.png",
            }
          }
          style={
            Object {
              "borderRadius": 10,
              "height": 200,
              "width": 200,
            }
          } />
      </View>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
            },
            Object {
              "paddingRight": 10,
            },
          ]
        }
        testID={undefined}>
        <Image
          source={
            Object {
              "uri": "https://image2.png",
            }
          }
          style={
            Object {
              "borderRadius": 10,
              "height": 200,
              "width": 200,
            }
          } />
      </View>
    </ScrollView>
    <View
      style={
        Object {
          "borderColor": "#efeff4",
          "borderWidth": 10,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#679fad",
            "fontSize": 20,
            "fontWeight": "bold",
            "padding": 10,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 18,
            "padding": 10,
            "paddingTop": 10,
          }
        } />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 30,
            "fontWeight": "bold",
            "padding": 10,
          }
        }>
        Associated Links
      </Text>
      <View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#679fad",
                "fontSize": 20,
                "fontWeight": "bold",
                "padding": 10,
              }
            }>
            https://link1.html
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#679fad",
                "fontSize": 20,
                "fontWeight": "bold",
                "padding": 10,
              }
            }>
            https://link2.html
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              undefined,
            ]
          }
          testID={undefined}>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#679fad",
                "fontSize": 20,
                "fontWeight": "bold",
                "padding": 10,
              }
            }>
            https://link3.html
          </Text>
        </View>
      </View>
    </View>
  </ScrollView>
</Modal>
`;
